### PR TITLE
Improvements for HAL skeleton and Watchdog integration

### PR DESCRIPTION
**Pull request by SteelPh0enix**
- HAL skeleton refactored, cleaned up and improved
- Watchdog integrated with HAL and Aerugo
- Added proper hardware initialization procedure

### CODE REVIEW

**Review by Glamhoth — CHANGES_REQUESTED** [review]

File: arch/cortex-m/samv71-hal/src/drivers/watchdog/mod.rs [hunk 1]
```diff
@@ -72,6 +41,23 @@ impl Watchdog {
         }
     }
 
+    /// Converts duration to watchdog counter value
+    fn convert_duration_to_counter_value(duration: MillisDurationU32) -> u16 {
```
[the remark this answers is not in the document]
Comment (SteelPh0enix) [reply arch/cortex-m/samv71-hal/src/drivers/watchdog/mod.rs]: moved

File: arch/cortex-m/samv71-hal/src/hal.rs [hunk 2]
```diff
@@ -1,50 +1,109 @@
...
     /// Frequency for the time types (TODO)
     const TIMER_FREQ: u32 = 1_000_000;
 
-    /// Create new HAL instance.
-    pub fn new(peripherals: Peripherals) -> Self {
-        Hal {
-            peripherals: InternalCell::new(Some(peripherals)),
-        }
+    /// Create new HAL instance from PAC peripherals.
+    /// SAFETY: This function is safe to call only once.
```
[the remark this answers is not in the document]
Comment (SteelPh0enix) [reply arch/cortex-m/samv71-hal/src/hal.rs]: changed `/// SAFETY:` to markdown headers

File: arch/cortex-m/samv71-hal/src/hal.rs [hunk 3]
```diff
@@ -1,50 +1,109 @@
...
+    /// Subsequent calls will return an error, indicating that HAL instance has already been created.
+    pub fn new() -> Result<Self, HalError> {
+        HAL_CREATION_LOCK.lock(|lock| {
+            if *lock {
+                return Err(HalError::HalAlreadyCreated);
+            }
+
+            *lock = true;
```
[the remark this answers is not in the document]
Comment (SteelPh0enix) [reply arch/cortex-m/samv71-hal/src/hal.rs:46]: refactored to use `?`

File: arch/cortex-m/samv71-hal/src/drivers/watchdog/config.rs
[the remark this answers is not in the document]
Comment (SteelPh0enix) [reply arch/cortex-m/samv71-hal/src/drivers/watchdog/config.rs:1]: moved

File: arch/cortex-m/samv71-hal/src/drivers/watchdog/mod.rs
[the remark this answers is not in the document]
Comment (SteelPh0enix) [reply arch/cortex-m/samv71-hal/src/drivers/watchdog/mod.rs:1]: moved

File: arch/cortex-m/samv71-hal/src/drivers/watchdog.rs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+/// # Safety
+/// Watchdog does not auto-implement Sync due to WDT structure containing a pointer.
+/// Since it owns WDT, and it's running in single-core environment, it's safe to share.
+unsafe impl Sync for Watchdog {}
+
+impl Watchdog {
+    /// Create a watchdog instance from PAC peripheral.
+    ///
+    /// # Arguments
```
Comment (Glamhoth) [inline arch/cortex-m/samv71-hal/src/drivers/watchdog.rs:39]: `Parameters` to be inline with the rest of the docs

**Review by Glamhoth — CHANGES_REQUESTED** [review]

File: arch/cortex-m/samv71-hal/src/drivers/watchdog.rs (hunk 4 above)
Comment (SteelPh0enix) [reply arch/cortex-m/samv71-hal/src/drivers/watchdog.rs]: renamed in whole project

**Pull request merged**